mesos_collectd.py

```diff
@@ -79,9 +80,13 @@ def configure_callback(conf, is_master, prefix, cluster, instance, path, host,
 
     global MESOS_VERSION
     binary = '%s/%s' % (path, 'mesos-master' if is_master else 'mesos-slave')
-    # Expected output: mesos <version_string>
-    version = subprocess.check_output([binary, '--version'])
-    MESOS_VERSION = version.strip().split()[-1]
+    if os.path.exists(binary):
+        # Expected output: mesos <version_string>
+        version = subprocess.check_output([binary, '--version'])
+        MESOS_VERSION = version.strip().split()[-1]
+    else:
+        version = get_version_from_api(host, port)
```

Comment:
great question. my opinion was that there are less things that would go wrong with getting version from the binary and the majority of deployments might be on mesos system.  So using the version api was not as reliable but useful in this case of being on remote system.